cdap-security/src/main/java/io/cdap/cdap/security/auth/context/AuthenticationTestContext.java

```diff
@@ -25,6 +26,15 @@
 public class AuthenticationTestContext implements AuthenticationContext {
   @Override
   public Principal getPrincipal() {
-    return new Principal(System.getProperty("user.name"), Principal.PrincipalType.USER);
+    Credential credential = null;
+    try {
+      String credentialValue = System.getProperty("user.credential.value");
+      Credential.CredentialType credentialType = Credential.CredentialType
+        .valueOf(System.getProperty("user.credential.type"));
+      credential = new Credential(credentialValue, credentialType);
+    } catch (NullPointerException e) {
+      // Skip creating a credential if properties were not set.
```

Comment:
Don't catch NPE and do no-op. Do explicit `if` checks on the properties instead.

Reply:
Since this is a cherry-pick, I've gone ahead and opened a new PR #13533 to address these comments.